src/server/controllers.js

```diff
@@ -0,0 +1,177 @@
+'use strict';
+
+const SUCCESS_RES_CODE = 200;
+const parse = require('yargs-parser');
+const {
+	listMetrics,
+	getRepositories,
+	generateDetailTable,
+} = require('../utils');
+
+const { getMetrics } = require('../metrics');
+
+// Metric names and their extraction method to be used on the query result (Order is preserved)
+const metricNames = [
```

Comment:
seems like this file contains a lot of copied code from elsewhere in the repo?

Reply:
Yes, it is the code from the `details.js` file. Kindly consider this commit as an intermediate step. I was having some trouble with the exports(for some reason importing the `detail` function gave errors, saying detail wasn't a function). So I just copy-pasted the code instead to make things work for the **time being**. I will fix this in a future commit and perhaps squash this commit.

Reply:
Sounds good :-) just calling it out so it's not forgotten.

Reply:
This is still a doubt for me, could you probably suggest me a fix for this? I'm not sure why importing from detail.js is not working. Perhaps there is some issue with express js? Not really sure

Reply:
I'm not sure either. Can you restore the import, and i'll try it locally?

Reply:
Let's address this issue at the very end of this PR, once we have everything in working order.